src/Tribe/Attendees.php

```diff
@@ -790,34 +804,95 @@ public function maybe_generate_csv() {
 		 */
 		$items = apply_filters( 'tribe_events_tickets_attendees_csv_items', $items, $event_id );
 
-		if ( ! empty( $items ) ) {
-			$charset  = get_option( 'blog_charset' );
-			$filename = sanitize_file_name( $event->post_title . '-' . __( 'attendees', 'event-tickets' ) );
-
-			// Output headers so that the file is downloaded rather than displayed.
-			header( "Content-Type: text/csv; charset=$charset" );
-			header( "Content-Disposition: attachment; filename=$filename.csv" );
-
-			// Create the file pointer connected to the output stream.
-			$output = fopen( 'php://output', 'w' );
-
-			/**
-			 * Allow filtering the field delimiter used in the CSV export file.
-			 *
-			 * @since 5.1.3
-			 *
-			 * @param string $delimiter The field delimiter used in the CSV export file.
-			 */
-			$delimiter = apply_filters( 'tribe_tickets_attendees_csv_export_delimiter', ',' );
-
-			// Output the lines into the file.
-			foreach ( $items as $item ) {
-				fputcsv( $output, $item, $delimiter );
-			}
+		if ( empty( $items ) ) {
+			return;
+		}
 
-			fclose( $output );
-			exit;
+		$charset  = get_option( 'blog_charset' );
+		$filename = sanitize_file_name( $event->post_title . '-' . __( 'attendees', 'event-tickets' ) );
+
+		// Output headers so that the file is downloaded rather than displayed.
+		header( "Content-Type: text/csv; charset={$charset}" );
+		header( "Content-Disposition: attachment; filename={$filename}.csv" );
+
+		// Create the file pointer connected to the output stream.
+		$output = fopen( 'php://output', 'w' );
+
+		/**
+		 * Allow filtering the field delimiter used in the CSV export file.
+		 *
+		 * @since 5.1.3
+		 *
+		 * @param string $delimiter The field delimiter used in the CSV export file.
+		 */
+		$delimiter = apply_filters( 'tribe_tickets_attendees_csv_export_delimiter', ',' );
+
+		// Output the lines into the file.
+		foreach ( $items as $item ) {
+			fputcsv( $output, $item, $delimiter ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.file_ops_fputcsv
 		}
+
+		fclose( $output );
+		exit;
+	}
+
+	/**
+	 * Export a CSV containing all attendees.
+	 *
+	 * Called when an Admin requested a CSV export from the attendees table. Generates
+	 * the download file and finishes the execution.
+	 * 
+	 * @since TBD
+	 */
+	public function export_csv_for_all() {
+		if ( empty( $_GET['attendees_csv'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return;
+		}
+
+		if ( empty( $_GET['attendees_csv_nonce'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return;
+		}
+
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		// Generate filtered list of attendees.
+		$items = $this->generate_filtered_list( 'all' );
+
+		// Sanitize items for CSV usage.
+		$items = $this->sanitize_csv_rows( $items );
+
+		if ( empty( $items ) ) {
+			return;
+		}
+
+		$charset  = get_option( 'blog_charset' );
```

Comment:
Similar to the above  comment, we have two methods that are doing almost the same thing: building a list of attendees and creating a file download.

It would be nicer to see a single method that accepts `$items` as a parameter, and generates the CSV with those items. Then each of the methods could simply handle how `$items` is created and pass that to the new method.